### PR TITLE
loosely document supported architectures

### PR DESCRIPTION
**Pull request by nomius10**
This is an attempt at tackling #1371, solving it only partially.

It adds a table that tracks which binaries compile on a couple of targets, as well as a python script to generate said table. The target names are taken from `rustup target --list`. The script runs `cargo check` on stable, and if no errors occur, the cell is marked with `y`.

This provides a general overview of the state of cross-platform compatibility.

- add script to generate a table of bins that compile
- add section in README

I'm open to comments / suggestions

### CODE REVIEW

**Comment by ArniDagur** [comment]
I think it would be more readable if you sorted the rows and columns by how many `y`'s there are in each one. More supported architectures would be nearer the top, and tools with more support are more to the left. I hope that makes sense.

File: README.md [hunk 1]
```diff
@@ -374,6 +374,33 @@ Utilities
...
+|---------------------------|----|------|------|--------|---|-----|-----|-----|------|-----|----|--|------|---|----|--|---------|-------|--|----|---|------|----|------|-----|---|----|------|-------|----|------|--------|--|-------|----|----|----|--|-------|--|-----|------|-----|------|----|--|----|--|-----|-----|------|--|-----|-------|-----|--------|------|---|---|--------|--------|-------|--|-----|---|-----|----|-----|----|-----|----|------|---|----|---|----|---|----|-------|-----|--|----|--------|-----|---|-----|--------|----|------|------|-----|--|---|------|---|
+|          aarch64-apple-ios|    |      |      |        |   |     |     |     |      |     |    |  |      |   |    |  |         |       |  |    |   |      |    |      |     |   |    |      |       |    |      |        |  |       |    |    |    |  |       |  |     |      |     |      |    |  |    |  |     |     |      |  |     |       |     |        |      |   |   |        |        |       |  |     |   |     |    |     |    |     |    |      |   |    |   |    |   |    |       |     |  |    |        |     |   |     |        |    |      |      |     |  |   |      |   |
```
Comment (Arcterus) [inline README.md:384]: Are these entirely empty rows due to missing `std` for this platform on your system.

Comment (nomius10) [reply README.md:384]: Yes, it seems I'm missing the SDK

```
└$ cargo check -p uu_nohup --bin nohup --target=aarch64-apple-ios
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `rustc - --crate-name ___ --print=file-names --target aarch64-apple-ios --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=cfg` (exit code: 1)
--- stderr
error: Error loading target specification: failed to get iphoneos SDK path: No such file or directory (os error 2). Use `--print target-list` for a list of built-in targets
```

Comment (nomius10) [reply README.md:384]: I thought `rustup target add` would suffice for a simple check

Comment (Arcterus) [reply README.md:384]: It work for most (common) targets, but not all.  You can try to parse the error message for stuff like this (although this kinda jank), keep a list of targets that work, or try to set up the missing requirements for the broken targets in the script.  I don’t have a strong preference for which, so it’s up to you.

**Comment by sylvestre** [comment]
@nomius10 I see that it is still marked as draft. Are you happy to land it or not?

**Comment by nomius10** [comment]
@sylvestre Hey! I'd like to implement some checks regarding certain tests (e.g., as Arcterus pointed out, IOS checks should validate that the SDK is installed). I'll look into that later today.

**Comment by nomius10** [comment]
- added a few checks.
- made apple targets to show as "apple MacOS"/"apple IOS"
- removed the IOS lines from the table as I don't own a mac...

**Comment by sylvestre** [comment]
Let's take it, we can always iterate later!

**Comment by sylvestre** [comment]
@nomius10 Hello,
I am trying to rephrase the csv and table, how do you do that? thanks

**Comment by nomius10** [comment]
@sylvestre 
How embarrassing, I forgot to document the script in any shape or form.

1. cd into the `docs` folder
2. run `python compiles_table.py render` - this prints the table in a markdown format, from a CSV cache
3. run `python compiles_table.py check aarch64-unknown-linux-gnu x86_64-unknown-linux-gnu [...]` to update a row in the CSV cache. You can check against as many toolchains as you want, only condition is that you have to have the toolchain installed, otherwise an assertion fails and the script stops.
4. rerender the table and manually copy the markdown over

You need the `tqdm` python package installed

**Comment by sylvestre** [comment]
Thanks. I pushed an update before seeing your comment:
https://github.com/uutils/coreutils/commit/7af3327c5c730273dd05fcc882afbac15cae81a7

I will double check the script, I think fuchsia should be supported :)

**Pull request merged**